src/core/net/ip4_types.cpp

```diff
@@ -148,6 +147,46 @@ Address::InfoString Address::ToString(void) const
     return string;
 }
 
+Error Cidr::FromString(const char *aString)
+{
+    constexpr char     kSlashChar     = '/';
+    constexpr uint16_t kMaxCidrLength = 32;
+
+    Error error = kErrorParse;
+
+    SuccessOrExit(AsCoreType(&mAddress).FromString(aString, kSlashChar));
+
+    aString = StringFind(aString, kSlashChar);
+    VerifyOrExit(aString != nullptr);
+    aString++;
+
+    {
+        uint8_t  hasFirstDigit = false;
+        uint16_t value         = 0;
+
+        for (char digitChar = *aString;; ++aString, digitChar = *aString)
+        {
+            if ((digitChar < '0') || (digitChar > '9'))
+            {
+                break;
+            }
+
+            value = static_cast<uint16_t>((value * 10) + static_cast<uint8_t>(digitChar - '0'));
+            VerifyOrExit(value <= kMaxCidrLength);
+            hasFirstDigit = true;
+        }
```

Comment:
We has similar code in `Address::FromString()` reading an int value?
Maybe we can define a helper method/function for this (in `string.hpp` maybe so even more general and can be used by all modules)?

Reply:
Would expect another PR for this.